src/charm.py

```diff
@@ -36,6 +37,7 @@ def __init__(self, *args: typing.Any):
         self.on.define_event("run", RunEvent)
 
         self.image_observer = image.Observer(self)
+        self.cos_observer = cos.Observer(self)
```

Comment:
Is there a specific reason on why we don't directly instantiate the cosAgent class here but instead make a wrapper class for it?